samtranslator/model/sam_resources.py

```diff
@@ -96,37 +96,37 @@ class SamFunction(SamResourceMacro):
         "ImageUri": PropertyType(False, is_str()),
         "PackageType": PropertyType(False, is_str()),
         "InlineCode": PropertyType(False, one_of(is_str(), is_type(dict))),
-        "DeadLetterQueue": PropertyType(False, is_type(dict)),
+        "DeadLetterQueue": PropertyType.optional_dict(),
         "Description": PropertyType(False, is_str()),
         "MemorySize": PropertyType(False, is_type(int)),
         "Timeout": PropertyType(False, is_type(int)),
-        "VpcConfig": PropertyType(False, is_type(dict)),
+        "VpcConfig": PropertyType.optional_dict(),
         "Role": PropertyType(False, is_str()),
-        "AssumeRolePolicyDocument": PropertyType(False, is_type(dict)),
+        "AssumeRolePolicyDocument": PropertyType.optional_dict(),
         "Policies": PropertyType(False, one_of(is_str(), is_type(dict), list_of(one_of(is_str(), is_type(dict))))),
         "RolePath": PassThroughProperty(False),
         "PermissionsBoundary": PropertyType(False, is_str()),
         "Environment": PropertyType(False, dict_of(is_str(), is_type(dict))),
         "Events": PropertyType(False, dict_of(is_str(), is_type(dict))),
-        "Tags": PropertyType(False, is_type(dict)),
+        "Tags": PropertyType.optional_dict(),
```

Comment:
I find this harder to read.

Now we have a mix of `PropertyType(... is_type(dict))`, `PropertyType.optional_dict()`, etc. It's clear what this all is, why pick one over there, or why optional dicts are special.

Reply:
yeah, need a follow up to replace others